### PR TITLE
Feature/orb read datasets

### LINKED ISSUE #273: Policy Dataset read

### PR DESCRIPTION
**Pull request by mclcavalcante**
Resolves #273

### CODE REVIEW

**Review by dscabral — CHANGES_REQUESTED** [review]

**Review by dscabral — DISMISSED** [review]

File: policies/api/http/logging.go [hunk 1]
```diff
@@ -214,6 +214,48 @@ func (l loggingMiddleware) ValidateDataset(ctx context.Context, token string, d
...
+}
+
 func NewLoggingMiddleware(svc policies.Service, logger *zap.Logger) policies.Service {
 	return &loggingMiddleware{logger, svc}
 }
+
+func (l loggingMiddleware) ListDatasetByPolicyIDInternal(ctx context.Context, policyID string, ownerID string) (_ []policies.Dataset, err error) {
+	defer func(begin time.Time) {
+		if err != nil {
+			l.logger.Warn("method call: list_dataset_by_groups",
```
Comment (weyrick) [inline policies/api/http/logging.go:252]: should this be `list_dataset_by_policy_internal` as per the function name?

File: policies/api/http/logging.go [hunk 2]
```diff
@@ -214,6 +214,48 @@ func (l loggingMiddleware) ValidateDataset(ctx context.Context, token string, d
...
+				zap.Error(err),
+				zap.Duration("duration", time.Since(begin)))
+		} else {
+			l.logger.Info("method call: list_dataset_by_groups",
+				zap.Duration("duration", time.Since(begin)))
+		}
+	}(time.Now())
+	return l.svc.ListDatasetsByPolicyIDInternal(ctx, policyID, ownerID)
```
Comment (weyrick) [inline policies/api/http/logging.go:260]: this seems to be calling a different function, where  `Datasets` is plural with an s, seems like that maybe a typo, or at least an inconsistency?

File: policies/api/http/transport.go [hunk 3]
```diff
@@ -89,6 +89,16 @@ func MakeHandler(tracer opentracing.Tracer, svcName string, svc policies.Service
 		decodeView,
 		types.EncodeResponse,
 		opts...))
+	r.Get("/policies/dataset/:id", kithttp.NewServer(
+		kitot.TraceServer(tracer, "view_policy")(viewDatasetEndpoint(svc)),
```
Comment (weyrick) [inline policies/api/http/transport.go:93]: `view_dataset` instead of `view_policy`

File: policies/api/http/transport.go [hunk 4]
```diff
@@ -89,6 +89,16 @@ func MakeHandler(tracer opentracing.Tracer, svcName string, svc policies.Service
...
+		decodeView,
+		types.EncodeResponse,
+		opts...))
+	r.Get("/policies/dataset", kithttp.NewServer(
+		kitot.TraceServer(tracer, "list_dataset")(listDatasetEndpoint(svc)),
```
Comment (weyrick) [inline policies/api/http/transport.go:98]: `list_datasets` plural for consistency

File: policies/api/http/responses.go [hunk 5]
```diff
@@ -128,43 +161,27 @@ func (s datasetRes) Empty() bool {
...
 
-type policyValidateRes struct {
-	Name        string         `json:"name"`
-	Backend     string         `json:"backend"`
-	Description string         `json:"description"`
-	Tags        types.Tags     `json:"tags"`
-	Policy      types.Metadata `json:"policy"`
+type datasetPageRes struct {
+	pageRes
+	Dataset []datasetRes `json:"dataset"`
```
Comment (weyrick) [inline policies/api/http/responses.go:166]: `Datasets` plural for consistency with arrays

File: policies/policies.go [hunk 6]
```diff
@@ -86,6 +91,12 @@ type Service interface {
 
 	// ValidateDataset validates a new Dataset without saving it
 	ValidateDataset(ctx context.Context, token string, d Dataset) (Dataset, error)
+
+	// ViewDatasetByID retrieving dataset by id with token
+	ViewDatasetByID(ctx context.Context, token string, datasetID string) (Dataset, error)
+
+	// ListDataset retrieve a list of Dataset by owner
+	ListDataset(ctx context.Context, token string, pm PageMetadata) (PageDataset, error)
```
Comment (weyrick) [inline policies/policies.go:99]: I think this should also be plural `ListDatasets`

File: policies/policies.go [hunk 7]
```diff
@@ -126,4 +137,10 @@ type Repository interface {
 
 	//DeleteDataset delete a existing dataset by id by ownerID
 	DeleteDataset(ctx context.Context, ownerID string, dsID string) error
+
+	// RetrieveDatasetByID Retrieves dataset by id
+	RetrieveDatasetByID(ctx context.Context, datasetID string, ownerID string) (Dataset, error)
+
+	// RetrieveAllDatasetByOwner retrieves the subset of Datasets owned by the specified user
+	RetrieveAllDatasetByOwner(ctx context.Context, ownerID string, pm PageMetadata) (PageDataset, error)
```
Comment (weyrick) [inline policies/policies.go:145]: `RetrieveAllDatasetsByOwner`

**Review by weyrick — CHANGES_REQUESTED** [review]

**Review by weyrick — APPROVED** [review]

**Pull request merged**